File: ChibuzorTasks/bankapp/src/bankapp/BankMain.java
package bankapp.src.bankapp;

import java.util.Scanner;

public class BankMain {
    public static void main(String[] args) {
        Bank bank = new Bank("MyBank");
        Scanner scanner = new Scanner(System.in);
        boolean running = true;

        while (running) {
            System.out.println("\nWelcome to " + bank.getBankName());
            System.out.println("1. Create Account");
            System.out.println("2. Deposit Money");
            System.out.println("3. Transfer Money");
            System.out.println("4. Check Balance");
            System.out.println("5. Update PIN");
            System.out.println("6. Exit");
            System.out.print("Choose an option: ");

            int choice = scanner.nextInt();
            scanner.nextLine();

            switch (choice) {
                case 1:
                    createAccount(bank, scanner);
                    break;
                case 2:
                    depositMoney(bank, scanner);
                    break;
                case 3:
                    transferMoney(bank, scanner);
                    break;
                case 4:
                    checkBalance(bank, scanner);
                    break;
                case 5:
                    updatePin(bank, scanner);
                    break;
                case 6:
                    running = false;
                    System.out.println("Thank you for using " + bank.getBankName() + ". Goodbye!");
                    break;
                default:
                    System.out.println("Invalid option. Please try again.");
            }
        }

        scanner.close();
    }

    private static void createAccount(Bank bank, Scanner scanner) {
        System.out.print("Enter first name: ");
        String firstName = scanner.nextLine();
        System.out.print("Enter last name: ");
        String lastName = scanner.nextLine();
        System.out.print("Set your PIN: ");
        String pin = scanner.nextLine();

        bank.createAccount(firstName, lastName, pin);
        System.out.println("Account created successfully!");
    }

    private static void depositMoney(Bank bank, Scanner scanner) {
        System.out.print("Enter account number: ");
        String accountNumber = scanner.nextLine();
        System.out.print("Enter amount to deposit: ");
        double amount = scanner.nextDouble();
        scanner.nextLine();

        try {
            bank.deposit(accountNumber, amount);
            System.out.println("Deposit successful. New balance: " + bank.checkBalance(accountNumber, "pin"));
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static void transferMoney(Bank bank, Scanner scanner) {
        System.out.print("Enter your account number: ");
        String senderAccount = scanner.nextLine();
        System.out.print("Enter recipient's account number: ");
        String recipientAccount = scanner.nextLine();
        System.out.print("Enter amount to transfer: ");
        double amount = scanner.nextDouble();
        scanner.nextLine();
        System.out.print("Enter your PIN: ");
        String pin = scanner.nextLine();

        try {
            bank.transfer(senderAccount, recipientAccount, amount, pin);
            System.out.println("Transfer successful.");
            System.out.println("Your new balance: " + bank.checkBalance(senderAccount, pin));
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static void checkBalance(Bank bank, Scanner scanner) {
        System.out.print("Enter account number: ");
        String accountNumber = scanner.nextLine();
        System.out.print("Enter your PIN: ");
        String pin = scanner.nextLine();

        try {
            double balance = bank.checkBalance(accountNumber, pin);
            System.out.println("Your balance: " + balance);
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static void updatePin(Bank bank, Scanner scanner) {
        System.out.print("Enter account number: ");
        String accountNumber = scanner.nextLine();
        System.out.print("Enter your current PIN: ");
        String oldPin = scanner.nextLine();
        System.out.print("Enter your new PIN: ");
        String newPin = scanner.nextLine();

        try {
            Account account = bank.getAccounts().stream()
                    .filter(a -> a.getAccountNumber().equals(accountNumber))
                    .findFirst()
                    .orElseThrow(() -> new IllegalArgumentException("Account not found"));

            account.updatePin(oldPin, newPin);
            System.out.println("PIN updated successfully.");
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: PythonClass/classWorks/Feb-2025/mybank/src/mybank/bank_main.py
from mybank.src.mybank.bank import Bank


def display_menu():
    print("\nWelcome to the Bank App!")
    print("1. Create Account")
    print("2. Deposit Money")
    print("3. Withdraw Money")
    print("4. Transfer Money")
    print("5. Check Balance")
    print("6. Exit")

def create_account(bank):
    first_name = input("Enter your first name: ")
    last_name = input("Enter your last name: ")
    pin = input("Set your PIN: ")
    try:
        bank.create_account(first_name, last_name, pin)
        print("Account created successfully!")
    except ValueError as e:
        print(f"Error: {e}")

def deposit_money(bank):
    account_number = input("Enter your account number: ")
    amount = float(input("Enter the amount to deposit: "))
    try:
        bank.deposit(account_number, amount)
        print("Deposit successful!")
    except ValueError as e:
        print(f"Error: {e}")

def withdraw_money(bank):
    account_number = input("Enter your account number: ")
    amount = float(input("Enter the amount to withdraw: "))
    pin = input("Enter your PIN: ")
    try:
        account = bank.find_account(account_number)
        account.withdraw(amount, pin)
        print("Withdrawal successful!")
    except ValueError as e:
        print(f"Error: {e}")

def transfer_money(bank):
    sender = input("Enter your account number: ")
    recipient = input("Enter the recipient's account number: ")
    amount = float(input("Enter the amount to transfer: "))
    pin = input("Enter your PIN: ")
    try:
        bank.transfer(sender, recipient, amount, pin)
        print("Transfer successful!")
    except ValueError as e:
        print(f"Error: {e}")

def check_balance(bank):
    account_number = input("Enter your account number: ")
    pin = input("Enter your PIN: ")
    try:
        balance = bank.check_balance(account_number, pin)
        print(f"Your balance is: {balance}")
    except ValueError as e:
        print(f"Error: {e}")

def main():
    bank = Bank("Python Bank")
    while True:
        display_menu()
        choice = input("Enter your choice: ")
        if choice == "1":
            create_account(bank)
        elif choice == "2":
            deposit_money(bank)
        elif choice == "3":
            withdraw_money(bank)
        elif choice == "4":
            transfer_money(bank)
        elif choice == "5":
            check_balance(bank)
        elif choice == "6":
            print("Thank you for using the Bank App. Goodbye!")
            break
        else:
            print("Invalid choice. Please try again.")
